### PR TITLE
Pester v5 compatability level

### PR DESCRIPTION
**Pull request by shaneis**
# A New PR

THANK YOU - We love to get PR's and really appreciate your time and help to improve this module

## Accepting a PR

Before we accept the PR - please confirm that you have run the tests locally to avoid our automated build and release process failing. You can see how to do that in our wiki

https://github.com/sqlcollaborative/dbachecks/wiki 

## Please confirm you have 0 failing Pester Tests

[] There are 0 failing Pester tests

## Changes this PR brings

Please add below the changes that this PR covers. It doesnt need an essay just the bullet points :-)

### CODE REVIEW

**Comment by shaneis** [comment]
@jpomfret Could you take a look at this PR when you can?
I got it to work but the changes seem a bit too much for my liking.

File: checks/Databasev5.Tests.ps1 [hunk 1]
```diff
@@ -50,7 +51,7 @@ BeforeDiscovery {
 
 
 
-Describe "Suspect Page" -Tag SuspectPage, High , Database -ForEach $InstancesToTest {
+Describe "Suspect Page" -Tags SuspectPage, High , Database -ForEach $InstancesToTest {
```
Comment (jpomfret) [inline checks/Databasev5.Tests.ps1:54]: These should all be `Tag` for the new version please :)

File: checks/Databasev5.Tests.ps1 [hunk 2]
```diff
@@ -179,7 +180,7 @@ Describe "Trustworthy Option" -Tag Trustworthy, DISA, Varied, CIS, Database -For
     }
 }
 
-Describe "Database Status" -Tag DatabaseStatus, High, Database -ForEach $InstancesToTest {
+Describe "Database Status" -Tags DatabaseStatus, High, Database -ForEach $InstancesToTest {
     $skip = ($__dbcconfig | Where-Object {$_.Name -eq 'skip.database.status' }).Value
```
Comment (jpomfret) [inline checks/Databasev5.Tests.ps1:184]: can you add a skip option to yours similar to this please

File: checks/Databasev5.Tests.ps1 [hunk 3]
```diff
@@ -1,5 +1,6 @@
+    $filename = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
 # So the v5 files need to be handled differently.
-# Ww will start with a BeforeDiscovery , $Filename which for the Database Checks will need to gather the Instances up front
+# Ww will start with a BeforeDiscovery , $Filename , $Filename which for the Database Checks will need to gather the Instances up front
```
Comment (jpomfret) [inline checks/Databasev5.Tests.ps1]: extra filename?

File: checks/Databasev5.Tests.ps1 [hunk 4]
```diff
@@ -193,4 +194,29 @@ Describe "Database Status" -Tag DatabaseStatus, High, Database -ForEach $Instanc
...
         }
     }
-}
\ No newline at end of file
+}
+
+Describe "Compatibility Level" -Tags CompatibilityLevel, High, Database -ForEach $InstancesToTest {
+    Context "Compatibility level matches server compatbility level" {
+        It "Database <_.Name> has the expected compatibility level on <_.SqlInstance>" -ForEach $psitem.Databases.Where{ if ($Database) { $_.Name -in $Database } else { $psitem.ConfigValues.statusexclude -notcontains $psitem.Name } } { 
+            Write-Host "$($Psitem | Select * | Format-List | OUt-string)"
```
Comment (jpomfret) [inline checks/Databasev5.Tests.ps1]: is this line just for testing?

File: checks/Databasev5.Tests.ps1 [hunk 5]
```diff
@@ -193,4 +194,29 @@ Describe "Database Status" -Tag DatabaseStatus, High, Database -ForEach $Instanc
             $psitem.Status | Should -Not -Match 'Suspect' -Because "We expect that there will be no databases in a Suspect state"
         }
     }
-}
\ No newline at end of file
+}
+
+Describe "Compatibility Level" -Tags CompatibilityLevel, High, Database -ForEach $InstancesToTest {
+    Context "Compatibility level matches server compatbility level" {
+        It "Database <_.Name> has the expected compatibility level on <_.SqlInstance>" -ForEach $psitem.Databases.Where{ if ($Database) { $_.Name -in $Database } else { $psitem.ConfigValues.statusexclude -notcontains $psitem.Name } } {
```
Comment (jpomfret) [inline checks/Databasev5.Tests.ps1:202]: then we'll add `-Skip:$skip` to the it as well

**Review by jpomfret — COMMENTED** [review]
Looks pretty good - there are quite a few changes that have to happen. 

For testing.. if you open the repo in the dev container..
you can then run this for your checks and confirm you are getting the same number of pass\skip\failed
`Invoke-PerfAndValidateCheck -Checks VirtualLogFile` 

image

Shout if you want some help with the dev container.. but it's pretty much open code... popup in bottom right that says 'shall we open the dev container' and then it'll start building it out (you need docker desktop running).

**Comment by SQLDBAWithABeard** [comment]
totally awesome @shaneis - Thank you - here is an updated branch that I will be merging into the Pesterv5 branch shortly once you folk have checked it is ok.

There is a description here

**Comment by SQLDBAWithABeard** [comment]
Lol - Seek closed your PR !!

File: checks/Databasev5.Tests.ps1 (hunk 1 above)
Comment (shaneis) [reply checks/Databasev5.Tests.ps1]: Updated!

File: checks/Databasev5.Tests.ps1 (hunk 2 above)
Comment (shaneis) [reply checks/Databasev5.Tests.ps1]: Added skip option

File: checks/Databasev5.Tests.ps1 (hunk 3 above)
Comment (shaneis) [reply checks/Databasev5.Tests.ps1]: Don't know how that happened - it showing up here, not the extra filename, that was caused by me fatfingering a copy/paste.
Thought I didn't push that though, updated

File: checks/Databasev5.Tests.ps1 (hunk 4 above)
Comment (shaneis) [reply checks/Databasev5.Tests.ps1]: Yup, and should be removed

File: checks/Databasev5.Tests.ps1 (hunk 5 above)
Comment (shaneis) [reply checks/Databasev5.Tests.ps1]: Added there

**Comment by shaneis** [comment]
Updated with the suggested changes there.
I'll work on the container tests now and let you know when it's finished 👍

**Pull request closed without merging**